src/statistical/unigram.jl

```diff
@@ -32,31 +30,24 @@ end
 use to initialize the SentencePieceModel by providing `vocab filepath`
 """    
 function load(path; unk_token="<unk>")
-    vocab = readlines(path)
-    vocabnew = split.(vocab, "\t")
-    voc = String[]
-    logprob = Float64[]
-    for i in 1:length(vocab)
-        vocab1 = vocabnew[i][1]
-        vocab1 = replace(vocab1, "▁"=>"_")
-        push!(voc, vocab1)
-        logp = vocabnew[i][2]
+    vocab_path = readlines(path)
+    vocabnlogp = split.(vocab_path, "\t")
+    vocab_map = Dict{String, Tuple{Float64, Int}}()
```

Comment:
We can replace the loop with
```suggestion
    vocab_map = Dict(tok=>parse(logp) for (tok, logp) in vocabnlogp)
``````

Reply:
vocab_map = Dict(tok=>(parse(Float64, logp), index) for (index, (tok, logp)) in enumerate(vocabnlogp)) 
got it